spec/services/preserved_object_handler_confirm_version_spec.rb

```diff
@@ -0,0 +1,335 @@
+              expect(pc.reload.size).to eq orig
+            end
+          end
+        end
+        it 'PreservedObject is not updated' do
+          orig_timestamp = po.updated_at
+          po_handler.confirm_version
+          expect(po.reload.updated_at).to eq orig_timestamp
+        end
+        it "logs at info level" do
+          expect(Rails.logger).to receive(:log).with(Logger::INFO, version_matches_po_msg)
+          expect(Rails.logger).to receive(:log).with(Logger::INFO, version_matches_pc_msg)
+          expect(Rails.logger).to receive(:log).with(Logger::INFO, updated_pc_db_msg)
+          po_handler.confirm_version
+        end
+        context 'returns' do
+          let!(:results) { po_handler.confirm_version }
+
+          # results = [result1, result2]
+          # result1 = {response_code: msg}
+          # result2 = {response_code: msg}
+          it '3 results' do
+            expect(results).to be_an_instance_of Array
+            expect(results.size).to eq 3
+          end
+          it 'VERSION_MATCHES results' do
+            code = PreservedObjectHandlerResults::VERSION_MATCHES
+            expect(results).to include(a_hash_including(code => version_matches_pc_msg))
+            expect(results).to include(a_hash_including(code => version_matches_po_msg))
+          end
+          it 'UPDATED_DB_OBJECT PreservedCopy result' do
+            code = PreservedObjectHandlerResults::UPDATED_DB_OBJECT
+            expect(results).to include(a_hash_including(code => updated_pc_db_msg))
+          end
+        end
+      end
+
+      context 'incoming version does NOT match db version' do
+        let(:po_handler) { described_class.new(druid, 1, 666, ep) }
+        let(:exp_msg_prefix) { "PreservedObjectHandler(#{druid}, 1, 666, #{ep})" }
+        let(:unexpected_version_pc_msg) {
+          "#{exp_msg_prefix} incoming version (1) has unexpected relationship to PreservedCopy db version; ERROR!"
+        }
+        let(:updated_pc_db_status_msg) {
+          "#{exp_msg_prefix} PreservedCopy status changed from ok to expected_vers_not_found_on_storage"
+        }
+        let(:updated_pc_db_obj_msg) { "#{exp_msg_prefix} PreservedCopy db object updated" }
+
+        context 'PreservedCopy' do
+          context 'changed' do
+            it 'status to expected_vers_not_found_on_storage' do
+              expect(pc.status).to eq PreservedCopy::OK_STATUS
+              po_handler.confirm_version
+              expect(pc.reload.status).to eq PreservedCopy::EXPECTED_VERS_NOT_FOUND_ON_STORAGE_STATUS
+            end
+            it 'last_audited' do
+              orig = pc.last_audited
+              po_handler.confirm_version
+              expect(pc.reload.last_audited).not_to eq orig
+            end
+            it 'last_checked_on_storage' do
+              orig = pc.last_checked_on_storage
+              po_handler.confirm_version
+              expect(pc.reload.last_checked_on_storage).not_to eq orig
```

Comment:
i think i'd have expected `.to be > orig`, as in the subsequent tests (ditto the test before this one)